source/fundamentals/connection/tls.txt

```diff
@@ -0,0 +1,296 @@
+method and pass the contents of your certificates in the options parameter.
+This method returns a ``SecureContext`` object that you can set as an option 
+in your ``MongoClientOptions`` object.
+
+The following code shows how to create a ``SecureContext`` object by
+calling the ``tls.createSecureContext()`` method and set the
+``secureContext`` client option:
+
+.. code-block:: js
+   :emphasize-lines: 2-6, 10
+   
+   // Create a SecureContext object
+   var sc = tls.createSecureContext({
+     ca: fs.readFileSync(`<path to CA certificate>`),
+     cert: fs.readFileSync(`<path to client certificate>`),
+     key: fs.readFileSync(`<path to client certificate>`),
+   });
+   
+   // Pass the SecureContext as a client option
+   const client = new MongoClient(uri, {
+     secureContext: sc,
+   });
+
+To learn more about the ``tls.createSecureContext()`` method and the
+``tls`` package, see the `Node.js TLS API documentation
+<https://nodejs.org/api/tls.html#tlscreatesecurecontextoptions>`__.
+
+For a runnable example that uses the ``secureContext`` client option, see
+the :ref:`SecureContext Example <node-securecontext-full-example>`.
+
+Provide Certificate Filepaths
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+You can include the filepaths for your certificates as client options to
+retrieve your certificates while connecting with TLS.
+
+The following code shows how to specify the ``tlsCAFile`` and
+``tlsCertificateKeyFile`` client options with certificate filepaths: 
+
+.. code-block:: js
+   :emphasize-lines: 3-4
+   
+   // Pass filepaths to client options
+   const client = new MongoClient(uri, {
+     tlsCAFile: `<path to CA certificate>`,
+     tlsCertificateKeyFile: `<path to client certificate>`,
+   });
+
+Create ``Buffer`` Objects to Store Certificates
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+You can pass the contents of your certificate files as ``Buffer``
+objects in your client options to connect with TLS.
+
+The following code shows how to read the contents of your certificate
+files and pass the resulting ``Buffer`` objects to the ``ca``, ``cert``, and
+``key`` client options:
+
+.. code-block:: js
+   :emphasize-lines: 2-4, 8-10
+   
+   // Read file contents
+   const ca = fs.readFileSync(`<path to CA certificate>`);
+   const cert = fs.readFileSync(`<path to client certificate>`);
+   const key = fs.readFileSync(`<path to client certificate>`);
+
+   // Pass Buffers to client options
+   const client = new MongoClient(uri, {
+     ca: ca,
+     cert: cert,
+     key: key,
+   });
+
+.. _node-securecontext-full-example:
+
+``SecureContext`` Example
+-------------------------
+
+This example shows how to create a ``SecureContext`` object and
+a ``MongoClient`` instance that includes TLS options. The example
+connects to MongoDB and executes a find query:
+
+.. code-block:: js
+   
+   const { MongoClient } = require("mongodb");
+   const fs = require("fs");
+   const tls = require("tls");
```

Comment:
```suggestion
   import { MongoClient } from "mongodb";
   import * as fs from "fs";
   import * as tls from "tls";
```

I think we use import syntax in our examples now mostly right? This is just a best practices thing, most users know how to translate between the two but we can save them an extra step (TS users likely want to see ^ this format)

Reply:
should we have code examples in JS and TS? Or at least include a note that the example is in TS?

Reply:
I don't think it is necessary, we have this on the "How to Use the Usage Examples"

> All examples use ES module imports. You can [enable ES module imports](https://nodejs.org/api/esm.html#esm_enabling) by adding the following key-value pair to your package.json file:

https://www.mongodb.com/docs/drivers/node/current/usage-examples/#how-to-use-the-usage-examples

So we should stick to import syntax, it is not the default yet, but it is becoming the defacto standard for folks TS or JS, so we can just write code the modern way.